### PR TITLE
Add guide for upgrading to v3

### PR DESCRIPTION
**Pull request by lox**
This adds a guide from upgrading from v2 to v3. 

TODO:

* [x] Cover git-clean-flags change (added -x)
* [x] Cover interpolation
* [x] Annotations

### CODE REVIEW

File: pages/agent/upgrading_to_v3.md.erb [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+<%= toc %>
+
+## What's changed
+
+* [Pipeline Plugins](/docs/agents/plugins) for easily sharing functionality between pipelines
+* The bootstrap (run as a sub-process for every job) has moved from a [shell script](https://github.com/buildkite/agent/blob/2-6-stable/templates/bootstrap.sh) to a sub-command of the `buildkite-agent bootstrap`. This means it's written in golang and is much easier for us to maintain and distribute.
+* Agent meta-data has been renamed to "tags"
+* Built-in [Docker and Docker Compose support](/docs/builds/docker-containerized-builds) has been deprecated and moved to dedicated plugins, [docker-compose](https://github.com/buildkite-plugins/docker-compose-buildkite-plugin) and [docker](https://github.com/buildkite-plugins/docker-buildkite-plugin).
+* Variable interpolation in pipeline.yml
+* Hooks support in Windows
```
Comment (keithpitt) [inline pages/agent/upgrading_to_v3.md.erb]: Don't forget annotations!

File: pages/agent/upgrading_to_v3.md.erb [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+```yaml
+steps:
+  - label: ':hammer: Tests'
+    command: 'scripts/tests.sh'
+    plugins:
+      docker-compose#v1.7.0:
+        run: app
+```
+
+Note that at present these plugins are implemented in bash, so will only work on non-windows platforms. Windows users should continue to use the environment vars.
```
Comment (keithpitt) [inline pages/agent/upgrading_to_v3.md.erb]: Would plugins work if they ran the agent within a "Bash for Windows" shell?

File: pages/agent/upgrading_to_v3.md.erb [hunk 3]
```diff
@@ -0,0 +1,94 @@
+# Upgrading to Buildkite Agent v3
+
+The Buildkite Agent has changed a lot in v3, but we've tried to keep it as backwards compatible as possible. We'll cover what’s changed and how to upgrade to new agents.
+
+<%= toc %>
+
+## What's changed
+
+* [Pipeline Plugins](/docs/agents/plugins) for easily sharing functionality between pipelines
+* The bootstrap (run as a sub-process for every job) has moved from a [shell script](https://github.com/buildkite/agent/blob/2-6-stable/templates/bootstrap.sh) to a sub-command of the `buildkite-agent bootstrap`. This means it's written in golang and is much easier for us to maintain and distribute.
```
Comment (keithpitt) [inline pages/agent/upgrading_to_v3.md.erb]: Also means most features work across both linux and windows.

File: pages/agent/upgrading_to_v3.md.erb (hunk 2 above)
Comment (lox) [reply pages/agent/upgrading_to_v3.md.erb]: Not without some changes, the agent currently checks operating system and uses that to switch between interpreters, batch for windows and shell for linux/unix. We could have an `interpreter` config that let's people make that choice independent of OS, or we could do some smarter detection vs just the OS switch.

File: pages/agent/upgrading_to_v3.md.erb (hunk 3 above)
Comment (lox) [reply pages/agent/upgrading_to_v3.md.erb]: 👍

**Comment by lox** [comment]
Ok, changes made @keithpitt.

**Comment by lox** [comment]
@harrietgrace we are going to get this into prod and then some copy iterating on it!

**Pull request merged**